apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-welcome-tour/src/tour-card.js

```diff
@@ -7,98 +7,201 @@ import './public-path';
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+
+// eslint-disable-next-line react-hooks/exhaustive-deps
+const useEffectOnlyOnce = ( func ) => useEffect( func, [] );
 
 function WelcomeTourCard( {
 	cardContent,
 	cardIndex,
+	justMaximized,
 	lastCardIndex,
 	onMinimize,
 	onDismiss,
-	setCurrentCard,
+	setJustMaximized,
+	setCurrentCardIndex,
 } ) {
 	const { description, heading, imgSrc } = cardContent;
+	const isLastCard = cardIndex === lastCardIndex;
+	// Ensure tracking is recorded once per slide view
+	useEffectOnlyOnce( () => {
+		// Don't track slide view if returning from minimized state
+		if ( justMaximized ) {
+			setJustMaximized( false );
+			return;
+		}
+		recordTracksEvent( 'calypso_editor_wpcom_tour_slide_view', {
+			slide_number: cardIndex + 1,
+			is_last_slide: isLastCard,
+			slide_heading: heading,
+			is_gutenboarding: window.calypsoifyGutenberg?.isGutenboarding,
+		} );
+	} );
 
 	return (
 		<Card className="welcome-tour-card" isElevated>
-			<CardOverlayControls onDismiss={ onDismiss } onMinimize={ onMinimize } />
+			<CardOverlayControls
+				setJustMaximized={ setJustMaximized }
+				onDismiss={ onDismiss }
+				onMinimize={ onMinimize }
+				slideNumber={ cardIndex + 1 }
+			/>
 			<CardMedia>
 				<img alt="Editor Welcome Tour" src={ imgSrc } />
 			</CardMedia>
 			<CardBody>
 				<h2 className="welcome-tour-card__heading">{ heading }</h2>
 				<p className="welcome-tour-card__description">
 					{ description }
-					{ cardIndex === lastCardIndex ? (
+					{ isLastCard ? (
 						<Button
 							className="welcome-tour-card__description"
 							isTertiary
-							onClick={ () => setCurrentCard( 0 ) }
+							onClick={ () => setCurrentCardIndex( 0 ) }
 						>
 							Restart tour
 						</Button>
 					) : null }
 				</p>
 			</CardBody>
 			<CardFooter>
-				<PaginationControl
-					currentPage={ cardIndex }
-					numberOfPages={ lastCardIndex + 1 }
-					setCurrentPage={ setCurrentCard }
-				/>
-				<div>
-					{ cardIndex === 0 ? (
-						<Button isTertiary={ true } onClick={ () => onDismiss() }>
-							No thanks
-						</Button>
-					) : (
-						<Button isTertiary={ true } onClick={ () => setCurrentCard( cardIndex - 1 ) }>
-							Back
-						</Button>
-					) }
-					{ cardIndex < lastCardIndex ? (
-						<Button
-							className="welcome-tour-card__next-btn"
-							isPrimary={ true }
-							onClick={ () => setCurrentCard( cardIndex + 1 ) }
-						>
-							{ cardIndex === 0 ? "Let's start" : 'Next' }
-						</Button>
-					) : (
-						<Button
-							className="welcome-tour-card__next-btn"
-							isPrimary={ true }
-							onClick={ () => onDismiss() }
-						>
-							Done
-						</Button>
-					) }
-				</div>
+				{ isLastCard ? (
+					<TourRating></TourRating>
+				) : (
+					<CardNavigation
+						cardIndex={ cardIndex }
+						lastCardIndex={ lastCardIndex }
+						onDismiss={ () => onDismiss( 'close-btn' ) }
```

Comment:
When clicking on "No thanks", we're sending `action: close-btn`.

It's because we're passing down this anonymous function to the child components. The anonymous function always calls `onDismiss( 'close-btn' )` and doesn't accept any arguments, e.g., `'no-thanks-btn'`

We should pass the function down instead:

```suggestion
						onDismiss={ onDismiss }
```

Reply:
Yes, don't know what I was thinking there.  Thanks for catching!